crates/storage/trie.rs

```diff
@@ -178,9 +172,21 @@ mod test {
     };
     use tempdir::TempDir;
 
+    use self::libdbmx_db::LibmdbxTrieDb;
+
+    /// Creates a new trie based on a temporary DB
+    fn new_temp() -> Trie<LibmdbxTrieDb> {
```

Comment:
nit: We should rename it to `new_temp_trie`

Reply:
Or restore the `Trie::new_temp` feature-gated method

Reply:
Renamed and moved to `test_utils` to avoid repetition, as suggested.

> Or restore the Trie::new_temp feature-gated method

I tried that first, but using a concrete `TrieDB` in the generic implementation caused compile time issues. I think your `test_utils` idea is an elegant solution.